### ISSUE TITLE
Ambiguity in parsing [value in list]

### ISSUE THREAD

**Issue by JanekPo** [opencypher/openCypher#315]
Expression `[value in list]`  can be parsed as:

1. list comprehension with empty condition and empty projection sections
2.  literal list of single boolean value of `in` operator

This leads to ambigous result for the query:
```
with 1 as x 
return [x in [1,2]] as result
```

Result 1:
```
+--------+
| result |
+--------+
| [1,2]  |
+--------+
```

Result 2:
```
+--------------+
| result       |
+--------------+
| [true]       |
+--------------+
```

**Comment by jespersm** [comment]
My £0.02: According to the spec (version 9), regarding List Comprehensions: "Either the WHERE part, or the expression, can be omitted, if you only want to filter or map respectively."

As I read it, In that case, `[x in [1,2]]` can't never be a list comprehension (result 1), but always be result 2.

So possibly, it's just the Grammar that needs adjusting.

**Comment by thobe** [comment]
@jespersm I really like that way of thinking about it. There are a few (too many) things that are encoded as rules like that instead of being explicit in the grammar. This seems like something that would be straightforward to move to the grammar.

**Comment by jespersm** [comment]
Yes, and it would make the grammar a little closer to LALR(1), but still not quite there.
